[docs] Update `expo-github-action` guides and examples (#21385)
expo/expo: docs/pages/eas-update/github-actions.mdx

          fi

      - name: Checkout repository
        uses: actions/checkout@v2
        uses: actions/checkout@v3

      - name: Setup Node
        uses: actions/setup-node@v2
        uses: actions/setup-node@v3
        with:
          node-version: 16.x
          node-version: 18.x
          cache: yarn

      - name: Setup Expo
        uses: expo/expo-github-action@v7
      - name: Setup EAS
        uses: expo/expo-github-action@v8
        with:

>           expo-version: latest

Author: 🫡

          eas-version: latest
          token: ${{ secrets.EXPO_TOKEN }}

      - name: Find yarn cache
        id: yarn-cache-path
        run: echo "::set-output name=dir::$(yarn cache dir)"

      - name: Restore cache
        uses: actions/cache@v2
        with:
          path: ${{ steps.yarn-cache-path.outputs.dir }}
          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}

>           restore-keys: ${{ runner.os }}-yarn-

Author: I'm not sure why this was here in the first place, we also have `cache: yarn` set on `actions/setup-node`. So either this should be removed, or that cache property should be removed.

      - name: Install dependencies

>         run: yarn install --immutable

Author: This is not a valid flag for Yarn v1, only v2+ (which is not being installed in this action code)

Create a file path named **.github/workflows/preview.yml** at the root of your project.

</Step>

<Step label="2">

Inside **preview.yml**, copy and paste the following snippet:

```yaml
name: preview
on: pull_request

jobs:
  update:
    name: EAS Update
    runs-on: ubuntu-latest
    permissions:
      pull-requests: write
    steps:
      - name: Check for EXPO_TOKEN
        run: |
          if [ -z "${{ secrets.EXPO_TOKEN }}" ]; then
            echo "You must provide an EXPO_TOKEN secret linked to this project's Expo account in this repo's secrets. Learn more: https://docs.expo.dev/eas-update/github-actions"
            exit 1
          fi

      - name: Checkout repository
        uses: actions/checkout@v3

      - name: Setup Node
        uses: actions/setup-node@v3
        with:
          node-version: 18.x
          cache: yarn

      - name: Setup EAS
        uses: expo/expo-github-action@v8
        with:
          eas-version: latest
          token: ${{ secrets.EXPO_TOKEN }}

      - name: Install dependencies
        run: yarn install

      - name: Create preview
        uses: expo/expo-github-action/preview@v8
        with:
          command: eas update --auto
```

> Although the code above is similar to the code in the previous example, there are a few differences. In this action, we changed the workflow event (`on`) to run every time a pull request is opened or updated. In the `update` job, we still set up Node, Expo's GitHub Action, and the dependencies using GitHub action's built in cache. But, instead of running `eas update --auto` ourselves, we let the [preview subaction](https://github.com/expo/expo-github-action/tree/main/preview#readme) run it for us. This will add a comment to the pull request with basic information about the update, and a QR code to scan the update.

Reviewer: ```suggestion
Although the code above is similar to the example in the previous section, there are a few differences. In this action, we changed the workflow event (`on`) to run every time a pull request is opened or updated. In the `update` job, we still set up Node, Expo's GitHub Action, and the dependencies using GitHub action's built-in cache. But, instead of running `eas update --auto` ourselves, we let the [preview subaction](https://github.com/expo/expo-github-action/tree/main/preview#readme) run it for us. This will add a comment to the pull request with basic information about the update and a QR code to scan the update.
```

> Don't forget to add the `permissions` section to the job. This enables the job to add comments to the pull request.

</Step>

<Step label="3">

> Finally, just like the previous example, we need to give the script above permission to run by providing an `EXPO_TOKEN` environment variable.

Reviewer: ```suggestion
You must give the script above permission to run by providing an `EXPO_TOKEN` environment variable.
```

Since we're re-listing the steps, we don't have to mention the previous section.

> If you already have done this, you can skip this step. Only one valid `EXPO_TOKEN` is required to authenticate GitHub Actions with your Expo account.

Reviewer: ```suggestion
You can skip this step if you have already set up `EXPO_TOKEN` in the previous section. Only one valid `EXPO_TOKEN` is required to authenticate GitHub Actions with your Expo account.

If you haven't, you must give the script above permission to run by providing an `EXPO_TOKEN` environment variable.
```

Instead, we should mention here that the if they have already set it up in the previous section, they can skip the steps below.
